docs: add Bot resource to Terraform role

benarent opened the pull request:
Documentation for New Resource in Teleport 16, needed for Teleport 16 Cluster.

benarent committed:
Add Bot resource

New Resource in Teleport 16

zmb3 commented:
I fixed the PR title, but we should also fix the commit message. 

_add bot resource_ is ambiguous and sounds like you are implementing a new resource when this is actually just a docs update.

benarent committed:
Documentaion about bot resource permissions for Terraform role

Documentation for new Bot Resource permissions in Teleport 16 required for Terraform to work.

benarent committed:
Merge branch 'benarent/bots-doc' of https://github.com/gravitational/teleport into benarent/bots-doc

benarent commented:
> _add bot resource_ is ambiguous and sounds like you are implementing a new resource when this is actually just a docs update.

Thanks Zac. I've pushed a new message, should I add a change log item for this?  Prior to Teleport 16, Bots could be created without access to the bots resource.
